### DISCUSSION TITLE
Storing Raw Binary files?

### DISCUSSION THREAD

**Discussion by Narumiii** [nlohmann/json#2481] [Q&A] [answered]
Hello i am trying to save ASCII characters from a array into a json element but it gives me an exception due unicode charset "what():  [json.exception.type_error.316] invalid UTF-8 byte at index 2: 0x90" 

I am doing it like this

Here's the part (due too big, you can upload your own and generate your one here: https://www.dcode.fr/file-data) of the binary i am trying to store
MZ����������ÿÿ��¸�������@�����������������������������������������º��´�Í!¸�LÍ!This program cannot be run in DOS mode.

Here's how i am trying to store it:

std::string buffer = "MZ����������ÿÿ��¸�������@�����������������������������������������º��´�Í!¸�LÍ!This program cannot be run in DOS mode";
nlohmann::json j;
j["binary"] = buffer ;

Probably i am not doing it by the right way sorry if not i am new into this, any help is appreciated thank you.

**Comment by nlohmann** [comment, marked as answer]
The error occurs inside the `dump` function. That function can take a parameter `error_handler` that can be set to `ignore` to ignore UTF-8 errors, see https://nlohmann.github.io/json/api/basic_json/dump/

**Reply by Narumiii** [reply]
Thank you very helpful.
